components/event-service/internal/events/subscribed/events.go

```diff
@@ -36,13 +43,51 @@ type eventsClient struct {
 	triggerLister kneventinglister.TriggerLister
 }
 
+// waitForInformersSyncOrDie blocks until all informer caches are synced, or panics after a timeout.
+func waitForInformersSyncOrDie(f externalversions.SharedInformerFactory) {
+	ctx, cancel := context.WithTimeout(context.Background(), informerSyncTimeout)
+	defer cancel()
+
+	err := hasSynced(ctx, f.WaitForCacheSync)
+	if err != nil {
+		log.Fatalf("Error waiting for caches sync: %s", err)
+	}
+}
+
+type waitForCacheSyncFunc func(stopCh <-chan struct{}) map[reflect.Type]bool
+
+// hasSynced blocks until the given informer sync waiting function completes. It returns an error if the passed context
+// gets canceled.
+func hasSynced(ctx context.Context, fn waitForCacheSyncFunc) error {
+	// synced gets closed as soon as fn returns
+	synced := make(chan struct{})
+
+	// closing stopWait forces fn to return, which happens whenever ctx
+	// gets canceled
+	stopWait := make(chan struct{})
+	defer close(stopWait)
+
+	go func() {
+		fn(stopWait)
```

Comment:
do you know if fn (i.e `externalversions.SharedInformerFactory.WaitForCacheSync` ) will return an error ? and what will happen if it does? does it get propagated to `waitForInformersSyncOrDie` ?

Reply:
Actually the `WaitForCacheSync()` doesn't return any errors from the signature, also checking the implementation, it doesn't throw any errors to be caught.

Reply:
after tracing the code here https://github.com/kubernetes/client-go/blob/master/informers/factory.go#L158 which eventually calls this https://github.com/kubernetes/client-go/blob/4529075edf97f1129ef696f95a4bc67c4798f334/tools/cache/shared_informer.go#L254 I think you need to handle the boolean result stored in the `map[type]` . Because according to the last link, this is set to false when an error happens and the controller needs to exit.

Reply:
even though I think the only condition where it will be set to false is if `stopWait` was closed, but I'd say it's better to handle the false result from a contract POV regardless of the implementation internal.